### PR TITLE
Fix error handling of `<dtml-except>` and clean up `DT_Try`

### LINKED ISSUE #25: <dtml-except> breaks on Python 2
Traceback:
```python
src/Zope/src/App/special_dtml.py:203: in _exec
    result = render_blocks(self._v_blocks, ns)
.../DocumentTemplate-3.0b4-py2.7.egg/DocumentTemplate/_DocumentTemplate.py:148: in render_blocks
    render_blocks_(blocks, rendered, md)
.../DocumentTemplate-3.0b4-py2.7.egg/DocumentTemplate/_DocumentTemplate.py:249: in render_blocks_
    block = block(md)
.../DocumentTemplate-3.0b4-py2.7.egg/DocumentTemplate/DT_Try.py:142: in render
    return self.render_try_except(md)
.../DocumentTemplate-3.0b4-py2.7.egg/DocumentTemplate/DT_Try.py:171: in render_try_except
    traceback.print_exc(100, f)
/usr/lib64/python2.7/traceback.py:233: in print_exc
    print_exception(etype, value, tb, limit, file)
/usr/lib64/python2.7/traceback.py:124: in print_exception
    _print(file, 'Traceback (most recent call last):')
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

file = <_io.StringIO object at 0x7f8ef106e150>, str = 'Traceback (most recent call last):', terminator = '\n'

    def _print(file, str='', terminator='\n'):
>       file.write(str+terminator)
E       TypeError: unicode argument expected, got 'str'
```

Reason: The used `io.StringIO` in `DT_Try.py` expects `unicode` to be written to it, but `traceback.py` from standard lib insists to write `str`.

Possible solution: Use `cStringIO.StringIO` as it was before version 3.0a3 but only on Python 2.

### PR DESCRIPTION
**Pull request by sallner**
During cleanup other errors occurred which hindered writing tests, so this is one PR. Feel free to keep the commits in the merge. 
Fixes #25

### CODE REVIEW

File: src/DocumentTemplate/tests/test_DT_Try.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+    def test_DT_Try__Try__01(self):
+        """It renders a try block if no exception occurs."""
+        html = self.doc_class(
+            '<dtml-try>'
+            'Variable "name": '
+            '<dtml-var expr="str(100)">'
+            '</dtml-try>')
+        res = html()
+        expected = ('Variable "name": 100')
```
Comment (icemac) [inline src/DocumentTemplate/tests/test_DT_Try.py]: I think the parenthesis are not needed here and on some other places down the file.

File: src/DocumentTemplate/tests/test_DT_Try.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+            '<dtml-except LookupError>'
+            'Exception variable: '
+            '<dtml-var expr="str(110)">'
+            '</dtml-try>')
+        res = html()
+        expected = ('Exception variable: 110')
+        self.assertEqual(res, expected)
+
+    def test_DT_Try__Try__08(self):
+        """It raises if no exception handler matches."""
```
Comment (icemac) [inline src/DocumentTemplate/tests/test_DT_Try.py]: Maybe "It raises the occurred exception if …"?

File: src/DocumentTemplate/tests/test_DT_Try.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+            'Variable "name": '
+            '<dtml-var expr="not_defined">'
+            '<dtml-except KeyError>'
+            'Exception variable: '
+            '<dtml-var expr="str(100)">'
+            '<dtml-except AttributeError>'
+            'Exception variable: '
+            '<dtml-var expr="str(110)">'
+            '</dtml-try>')
+        self.assertRaises(NameError, html)
```
Comment (icemac) [inline src/DocumentTemplate/tests/test_DT_Try.py]: I'd prefer: (here and on the other places where `self.assertRaises` is used)

```python
self.assertRaises(NameError):
    html()
```

File: src/DocumentTemplate/tests/test_DT_Try.py [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+            '<dtml-except LookupError>'
+            'Exception variable: '
+            '<dtml-var expr="str(110)">'
+            '</dtml-try>')
+        res = html(ret='Return variable: 101')
+        expected = ('Return variable: 101')
+        self.assertEqual(res, expected)
+
+    def test_DT_Try__Try__10(self):
+        """It does not break with strings as exceptions."""
```
Comment (icemac) [inline src/DocumentTemplate/tests/test_DT_Try.py]: But it raises a well defined exception if doing so.

File: tox.ini [hunk 5]
```diff
@@ -30,7 +30,7 @@ commands =
     coverage combine
     coverage html
     coverage xml
-    coverage report --fail-under=67
+    coverage report --fail-under=71
```
Comment (icemac) [inline tox.ini]: Cool! 💯

File: src/DocumentTemplate/tests/test_DT_Try.py (hunk 4 above)
Comment (sallner) [reply src/DocumentTemplate/tests/test_DT_Try.py]: That is also true.

File: src/DocumentTemplate/tests/test_DT_Try.py (hunk 3 above)
Comment (sallner) [reply src/DocumentTemplate/tests/test_DT_Try.py]: I refactored it to 
```python
with self.assertRaises(NameError):
    html()
```

File: tox.ini (hunk 5 above)
Comment (sallner) [reply tox.ini]: :tada:

**Pull request merged**